api/app/controllers/webhook.controller.ts

```diff
@@ -174,109 +178,120 @@ async function processCommit(
+      const response = await axios.get(rawUrl, {
+        headers: { Accept: "application/vnd.github.v3.raw" },
+      });
+      const mdContent = response.data;
+      const jsonContent: BaseParsedMdContent =
+        parseMdToJSON<BaseParsedMdContent>(mdContent);
+      const transcript_by = jsonContent.transcript_by.toLowerCase();
+
+      function isTranscriptValid(transcript_by: string): boolean {
+        const lowerCaseTranscriptBy = transcript_by.toLowerCase();
+        return (
+          lowerCaseTranscriptBy.includes("tstbtc") &&
+          lowerCaseTranscriptBy.includes("--needs-review")
+        );
+      }
 
-    const transcriptHash = generateUniqueHash(jsonContent);
-    const totalWords = getTotalWords(jsonContent.body);
-    const content = jsonContent;
+      if (!jsonContent) {
+        throw new Error(
+          "Malformed data: transcript content might not be in the correct format"
+        );
+      }
 
-    if (!transcriptHash || !totalWords) {
-      throw new Error(
-        "Malformed data: transcript content might not be in the correct format"
-      );
-    }
+      const totalWords = getTotalWords(jsonContent.body);
+      const content = jsonContent;
 
-    const existingTranscript = await Transcript.findOne({
-      where: { transcriptUrl: rawUrl },
-    });
+      if (!totalWords) {
+        throw new Error(
+          "Malformed data: transcript content might not be in the correct format"
+        );
+      }
 
-    if (existingTranscript && type === "added") {
-      throw new Error("transcript already exists");
-    }
+      const existingTranscript = await Transcript.findOne({
+        where: { transcriptUrl: rawUrl },
+      });
 
-    if (!isTranscriptValid(transcript_by)) {
-      throw new Error(
-        "Transcript not from TSTBTC or does not need review - did not queue transcript"
-      );
-    }
+      if (existingTranscript && type === "added") {
+        throw new Error("transcript already exists");
+      }
 
-    const transcript: TranscriptAttributes = {
-      originalContent: {
-        ...content,
-        title: content.title.trim(),
-      },
-      content: content,
-      transcriptHash,
-      transcriptUrl: rawUrl,
-      status: TranscriptStatus.queued,
-      contentTotalWords: totalWords,
-    };
+      if (!isTranscriptValid(transcript_by)) {
+        throw new Error(
+          "Transcript not from TSTBTC or does not need review"
+        );
+      }
 
-    let transcriptData: any;
+      const transcript: TranscriptAttributes = {
+        originalContent: {
+          ...content,
+          title: content.title.trim(),
+        },
+        content: content,
+        transcriptHash: "",
+        transcriptUrl: rawUrl,
+        status: TranscriptStatus.queued,
+        contentTotalWords: totalWords,
+      };
+
+      let transcriptData: any;
+
+      async function cacheTranscript(transcriptData: any) {
```

Comment:
Let's make this a reusable function in utils for transcript. We're doing the same thing here https://github.com/bitcointranscripts/transcription-review-backend/blob/b5e7e32801a806d8ae2234f9b21ffccf6958b1e5/api/app/controllers/transcript.controller.ts#L58 from line 58 to line 78

Reply:
Yea, we raised in chats last week, I still left it here because it's a WIP, will move it out